sources/api/controllers/GroupController.ts

```diff
@@ -9,50 +9,62 @@ import {Group} from "../../model/entity/Group";
 import {User} from "../../model/entity/User";
 import {ApiException} from "../../utils/apiException";
 import {GroupView} from "../views/GroupView";
-import {GroupsToUsers, Role} from "../../model/entity/GroupsToUsers";
+import {Calendar} from "../../model/entity/Calendar";
+import {CalendarsToOwners} from "../../model/entity/CalendarsToOwners";
 
 
 //
-// Create a new Group
+// Create a new Group and its associated Calendar
 //
 export async function createGroup(req: Request, res: Response) {
     const requestingUser = getRequestingUser(req);
 
-    const groupName = (req.body.name || "").trim();
-    let groupMemberNames = req.body.members || [];
-    groupMemberNames = removeDuplicates(groupMemberNames);
-    groupMemberNames = removeIfExists(groupMemberNames, requestingUser.pseudo);
+    const groupName: string = req.body.name;
+    const groupDescription: string = req.body.description || "";
+    if (!groupName) {
+        throw new ApiException(400, "Bad request");
+    }
 
-    const conn = await DbConnection.getConnection();
-    const userRepository = conn.getRepository(User);
+    const groupMembersPseudos = makeGroupMemberNamesList(req.body.members || [], requestingUser.pseudo);
 
-    const userPromises = groupMemberNames.map((memberPseudo: string) => {
-        return getUser(userRepository, memberPseudo);
-    });
+    const conn = await DbConnection.getConnection();
 
-    let groupMembers: User[] = <any[]>(await Promise.all(userPromises));
+    // Make sure all members exist
+    const groupMembers = await getAllUsers(conn, groupMembersPseudos);
     groupMembers.push(requestingUser);
 
-    const group = new Group(groupName);
-    if (!group.isValid()) {
-        throw new ApiException(400, "Invalid fields in Group");
+    // Create Calendar
+    const calendar = new Calendar(groupName);
+    if (!calendar.isValid()) {
+        throw new ApiException(400, "Invalid fields for Calendar");
     }
+    const createdCalendar = await Calendar.createCalendar(conn, calendar, groupMembers);
 
-    const groupRepository = conn.getRepository(Group);
-    const groupsToUsersRepository = conn.getRepository(GroupsToUsers);
-    const createdGroup = await Group.createGroup(groupRepository, groupsToUsersRepository, group, groupMembers);
+    // Create Group
+    const group = new Group(groupDescription, createdCalendar);
+    const createdGroup = await conn.manager.save(group);
 
     return res.status(201).json({
         message: "Group created",
         group: GroupView.formatGroup(createdGroup)
     });
 }
 
-async function getUser(userRepository: Repository<User>, pseudo: string) : Promise<User> {
-    const user = await userRepository.findOne({pseudo: pseudo});
+function makeGroupMemberNamesList(members: string[], requestingUserPseudo: string): string[] {
+    members = removeDuplicates(members);
+    members = removeIfExists(members, requestingUserPseudo);
```

Comment:
Does this mean that we can only create groups for ourselves ? 
I don't know if we couldn't want a "superuser" to create groups for other people, for example in a society a group could be created for each team.
What do you think ?

Reply:
For now we will only take care of single users. Handling this kind of permissions right now woud be too much work.

However, this make me think that some people might want to be in a Group only to look a the Calendar and not be taken into account by group features (like finding the best time slot for an event).

We'll think about it